[new][author] Add html2pdf.js package

eKoopmans opened the pull request:
# Pull request template for new library

Pull request for issue: N/A
Related issue(s): #13114

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/eKoopmans/html2pdf.git
* NPM package url (optional): https://www.npmjs.com/package/html2pdf.js
* License and its reference: MIT
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 36
  * Count of stars: 594
  * Count of forks: 250
* NPM download stats (optional):
  * Downloads in the last week: 3790

## Essential checklist

* [x] I'm the author of this library
  * [x] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

eKoopmans commented:
Hi, I've removed the homepage field per the original build failure, but I'm not sure what to do about the current error: `"CDNJS_CACHE_USERNAME" secret not set!`. Any suggestions? Thanks!

eKoopmans commented:
If everything else looks good, I'm happy to resubmit the PR as a single commit. Just let me know, thanks!

extend1994 commented on ajax/libs/html2pdf.js/package.json:
```diff
@@ -0,0 +1,34 @@
+      {
+        "basePath": "dist",
+        "files": [
+          "*.js"
```
How about using `!(include|require){,**/*}`, assuming you will release files for a browser under dist, but not under `include` or `require` folder.

eKoopmans replied:
Sounds right - everything at the root of `dist` is packaged as UMD, for browser and anything else. The `include` and `require` folders are for Node/ES and should be excluded.

Can you point me to a reference for the globbing used in cdnjs? Just want to make sure I understand everything going on in `!(include|require){,**/*}` (and any other options). Thanks!

extend1994 replied:
Sure :) 
cdnjs uses [glob](https://www.npmjs.com/package/glob) package, here is reference: https://github.com/cdnjs/autoupdate/blob/master/autoupdate.js#L8. 
`!(include|require){,**/*}` will fetch all subfolder and files under `basePath` except for `include` or `require` folder.

eKoopmans commented:
I made two changes:

1. Fixed the files glob as [suggested](https://github.com/cdnjs/cdnjs/pull/13216#discussion_r250100615).
2. Created a GH-Pages homepage for the project, and added its link back in.

It's failing now because it's out of date. I'm not able to rebase (or squash) on my own, I've been doing this all through the GH web UI (not able to download the full repo right now). Any help? I can close and resubmit as a fresh branch if it helps.

Thanks!

extend1994 commented on ajax/libs/html2pdf.js/package.json:
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "html2pdf.js",
```
Would you like to use `html2pdf` as `name` which is the same as GitHub repo name?
(As usual, if the library uses git auto-update, we would recommend you use GitHub repo as `name`. But it's not necessary)

eKoopmans replied:
I changed the GH repo name to match NPM/CDNJS.

eKoopmans commented:
Hi @extend1994, thanks for all of your help through the process. And great catch with the earlier versions - yes, adding support for those would be great (though I don't expect there's much demand for anything before v0.8.0).

eKoopmans commented:
Amazing, thanks again @extend1994 for all the help! It's exciting to see it come up on the cdnjs search.

extend1994 commented:
No problem 😄
